packages/react/src/components/form/Fieldset/Fieldset.tsx

```diff
@@ -83,13 +83,15 @@ export const Fieldset = forwardRef<HTMLFieldSetElement, FieldsetProps>(
                 hideLegend && utilityclasses.visuallyHidden,
               )}
             >
-              {readOnly && (
-                <PadlockLockedFillIcon
-                  className={classes.padlock}
-                  aria-hidden
-                />
-              )}
-              {legend}
+              <span className={classes.legendContent}>
+                {readOnly && (
+                  <PadlockLockedFillIcon
+                    className={classes.padlock}
+                    aria-hidden
+                  />
+                )}
+                {legend}
+              </span>
```

Comment:
I don't think we need the span here? I think the `legend` can always have `inline-flex`

Reply:
Then it will override `display: contents`

Reply:
Why do we need `display: contents`? From my testing it seems like there is no unexpected gap when the `legend` always has `display: inline-flex`

Reply:
The gap issue is for the error element. For the legend, it solves antoher issue. Here is a demo: https://codesandbox.io/p/sandbox/quirky-heyrovsky-l9vrwf?file=%2Findex.html